### PR TITLE
Ignore fetch and lookup-by-key nodes for state update comparison [KVL-854]

### PR DESCRIPTION
**Pull request by miklos-da**
Summary of changes:
* Introduced flag `ignoreFetchAndLookupByKeyNodes` in `NormalizationSettings`.
* Made `StateUpdateComparison` agnostic to the presence of fetch and/or lookup-by-key nodes in a `TransactionAccepted` update in case the previous flag is enabled.

**Motivation:** When comparing historical ledger exports, earlier versions included fetch and lookup-by-key nodes in their events, hence, the generated updates are different from the ones generated by recent code. This PR will enable integrity checking to pass on exports generated by earlier versions of the Daml SDK.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala [hunk 1]
```diff
@@ -162,6 +177,26 @@ object ReadServiceStateUpdateComparison {
...
+    val filteredNodes = nodes.map {
+      case (nid, node: Node.NodeExercises[Nid, Cid]) =>
+        val filteredNode = node.copy(children = node.children.filter(nodes.contains))
+        (nid, filteredNode)
+      case keep => keep
+    }
+    val filteredRoots = tx.roots.filter(filteredNodes.contains)
+    val version = filteredRoots.iterator.foldLeft(TransactionVersion.minVersion)((acc, nodeId) =>
+      TransactionVersion.Ordering.max(acc, nodes(nodeId).version)
+    )
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: Isn't there an utility for that? I'd rather avoid replicating this logic here.

Comment (miklos-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: @remyhaemmerle-da Do you have a suggestion?

Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: This is KV specific.  I will rather let it here.

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: If there isn't already encapsulated logic to decide a transaction's version based on its nodes, I'd at least extract it as a separate method for improved readability.

**Review by fabiotudone-da — APPROVED** [review]

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: Daml-LF code assumes there is **only** the engine and the deserialization that creates transaction object. 
The logic to infer the version is nested deeply inside the engine.

Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: The question is: do you update the version in KV when you rid the transaction of its Fetch and Lookup Nodes ?

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala [hunk 2]
```diff
@@ -162,6 +177,26 @@ object ReadServiceStateUpdateComparison {
...
+    val nodes = tx.nodes.filter {
+      case (_, _: Node.NodeFetch[Cid] | _: Node.NodeLookupByKey[Cid]) => false
+      case _ => true
+    }
+    val filteredNodes = nodes.map {
+      case (nid, node: Node.NodeExercises[Nid, Cid]) =>
+        val filteredNode = node.copy(children = node.children.filter(nodes.contains))
+        (nid, filteredNode)
+      case keep => keep
+    }
```
Comment (remyhaemmerle-da) [inline ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:192]: Please make at least one of the two patern matching exhaustive, so we get warning when we add new type of node.

File: ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala (hunk 1 above)
Comment (miklos-da) [reply ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateComparison.scala:196]: No, we don't update it, we just set it to the transaction's version (see [here](https://github.com/digital-asset/daml/blob/ecd3110f488566c3ceef3c19c023e8c2e81c960a/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitter.scala#L355)).

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM.

**Pull request merged**